### ISSUE TITLE
[FEATURE] Add multilingual search capability to DCLAP text search

### ISSUE THREAD

**Issue by raindropQwQ** [NeptuneHub/AudioMuse-AI#585]
**Is your feature request related to a problem? Please describe.**
DCLAP text search currently only supports English queries. For users with non-English search keywords (Chinese, Japanese, Korean, etc.), text search returns essentially random results.

**Describe the solution you'd like**
I ran a small experiment: distilling a lightweight projection layer that maps GTE embeddings into DCLAP's existing audio embedding space.

Training code and model weights are available at [raindropQwQ/AudioMuse-AI-DCLAP-Multilingual](https://github.com/raindropQwQ/AudioMuse-AI-DCLAP-Multilingual).

The idea is simple — GTE already understands 75+ languages, and DCLAP's audio embeddings are already indexed. A small projection (768→512, ~1.3 MB ONNX) bridges the two, so non-English queries can search against the existing audio index without any re-indexing or major changes — just swap the text encoder behind the search box to GTE + a small projection layer.

Results based on the Song Describer dataset (100 tracks, text-to-audio Recall@1):

| Language | DCLAP RoBERTa (current) | GTE + projection |
|----------|:-----------------------:|:----------------:|
| English  | 14%                     | **21%**          |
| Chinese  | 1%                      | **17%**          |
| Japanese | 2%                      | **12%**          |

In my own library, Chinese queries have improved from completely unusable to a usable level.

**Implementation Note:** I'm not fully certain there's zero negative impact on English since my test set is small. A conservative approach would be to only apply the projection for non-English queries, falling back to RoBERTa for English. That said, in my limited testing, English performance was actually improved rather than negatively affected.

**Describe alternatives you've considered**
Translate non-English queries to English, then use DCLAP RoBERTa as-is. But as demonstrated in #553, MarianMT produces unreliable translations for CJK languages, and adding a translation step increases latency and complexity.

**Additional context**
- **No re-indexing needed:** Zero changes required for existing audio embeddings or indexing.
- **No extra memory overhead:** GTE is already loaded for lyrics, so this adds zero additional memory at runtime.
- **Extremely lightweight:** The projection ONNX file is only 1.3 MB.
- **Resources:** [raindropQwQ/AudioMuse-AI-DCLAP-Multilingual](https://github.com/raindropQwQ/AudioMuse-AI-DCLAP-Multilingual).
- The distillation process is simple: encode English captions with both DCLAP RoBERTa (teacher) and GTE (student), then train a small MLP to align them. 
Even training with English-only data partially enables multilingual search. Because of GTE's cross-lingual transfer, similar concepts in different languages already have similar internal representations. Adding multilingual training data further improves non-English performance without hurting English.

**Comment by NeptuneHub** [comment]
Hi and thanks for this, can you share more details ?

For example:
- can you easy raise a PR in order to enable me to do a fast test ?
- the new 1.3mb model can work alone on text totally removing the 500mb actual CLAP text embederr?
- Which text did you have used for the distillation? Can you re-link the dataset and write here just some example ?
- Can you confirm that all the material used comply with the actual AGPLv3 license of AudioMuse-AI?

In short if we can increase language supported without decrease the quality on English, decrease size of the model, and we are aligned with AGPLv3 license, why not ?

Looking forward for the PR to try and the response of the above quesiton.

**Comment by raindropQwQ** [comment]
Thanks for the response! Here are the answers:
1. PR: #597
2. Can the 1.3MB model work alone?
Not entirely alone — it works as a projection layer on top of the GTE model that's already bundled in the Docker image (gte-multilingual-base-int8.onnx, 325MB). The flow is:
Query text → GTE → 1.3MB projection → 512-dim DCLAP space
So it removes the 478MB DCLAP RoBERTa text model from the query path, reusing GTE instead.
3. Training data + examples:
- [MusicCaps](https://huggingface.co/datasets/google/MusicCaps) (5,521 captions, CC BY-SA 4.0) — e.g. "A low quality recording of a hip hop song with a male voice rapping over a beat with a bass and drums"
- [Song Describer Dataset](https://huggingface.co/datasets/renumics/song-describer-dataset) (746 captions, CC BY 4.0) — e.g. "A slow-paced pop ballad with gentle piano and soft female vocals"
- Captions translated to 14 languages via Gemini API for multilingual coverage
4. License compliance: ✅
- MusicCaps: CC BY-SA 4.0 (compatible with GPL per CC's official list (https://creativecommons.org/share-your-work/licensing-considerations/compatible-licenses/))
- [Song Describer](https://zenodo.org/records/10072001): CC BY-SA 4.0
- Projection weights published under GPL-3.0

**Comment by NeptuneHub** [comment]
Ok good, so I’ll wait your pr to be ready to test it. Also remember that in case I’ll merge it I’ll do a copy of your projection layer on a model specific release. This because I like having all the model used under my repository control. I don’t want that a third party delete the resource and compilation brake. Off course I’ll clearly state all the attributions.
Thanks!

**Comment by raindropQwQ** [comment]
Thank you for the reply. The PR (#597) is ready for testing, GTE + projection layer is now available for use. However, the old models have not been deleted yet, the warmup-loaded model has not been changed yet either.

Pre-built Docker images are available if you'd like to test directly:
- Intel/ARM
`docker pull ghcr.io/raindropqwq/audiomuse-ai:latest`
- NVIDIA
`docker pull ghcr.io/raindropqwq/audiomuse-ai:latest-nvidia`

**Comment by NeptuneHub** [comment]
Hi and thanks for your effort in this. I did some test comparing at least the english language and the result is not really predictable and so that comparable. I prefear to keep the as-is but really thanks for all your effort in this.
